pennylane/templates/tensornetworks/mps.py

```diff
@@ -17,68 +17,83 @@
         )
 
     if n_block_wires > n_wires:
         raise ValueError(
             f"n_block_wires must be smaller than or equal to the number of wires; got n_block_wires = {n_block_wires} and number of wires = {n_wires}"
         )
 
-    if n_wires % (n_block_wires / 2) > 0:
+    offset_limits = (
+        qml.math.array([-n_block_wires // 2 + 1, n_block_wires // 2 - 1]) + n_block_wires % 2
+    )
+    if offset < offset_limits[0] or offset > offset_limits[1]:
+        raise ValueError(
+            f"Provided offset is outside the expected range; the expected range for n_block_wires = {n_block_wires} is range{offset_limits[0], offset_limits[1]}"
+        )
+
+    if not abs(offset) and not n_block_wires % 2 and n_wires % (n_block_wires / 2) > 0:
         warnings.warn(
             f"The number of wires should be a multiple of {int(n_block_wires/2)}; got {n_wires}"
         )
 
+    n_step = n_block_wires // 2 + offset
+    n_layers = len(wires) - int(len(wires) % (n_block_wires // 2)) - n_step
+
     return tuple(
         tuple(wires[idx] for idx in range(j, j + n_block_wires))
         for j in range(
             0,
-            len(wires) - int(len(wires) % (n_block_wires // 2)) - n_block_wires // 2,
-            n_block_wires // 2,
+            n_layers,
+            n_step,
         )
+        if not j + n_block_wires > len(wires)
     )
 
 
 class MPS(Operation):
-    """The MPS template broadcasts an input circuit across many wires following the architecture of a Matrix Product State tensor network.
+    r"""The MPS template broadcasts an input circuit across many wires following the architecture of a Matrix Product State tensor network.
     The result is similar to the architecture in `arXiv:1803.11537 <https://arxiv.org/abs/1803.11537>`_.
 
-    The argument ``block`` is a user-defined quantum circuit.``block`` should have two arguments: ``weights`` and ``wires``.
-    For clarity, it is recommended to use a one-dimensional list or array for the block weights.
+    The keyword argument ``block`` is a user-defined quantum circuit that should accept two arguments: ``wires`` and ``weights``.
+    The latter argument is optional in case the implementation of ``block`` doesn't require any weights. Any additional arguments
+    should be provided using the ``kwargs``.
 
     Args:
         wires (Iterable): wires that the template acts on
         n_block_wires (int): number of wires per block
         block (Callable): quantum circuit that defines a block
         n_params_block (int): the number of parameters in a block; equal to the length of the ``weights`` argument in ``block``
         template_weights (Sequence): list containing the weights for all blocks
+        offset (int): offset for positioning the subsequent blocks relative to the default :math:`O_d = \lfloor \text{n_block_wires}/2  \rfloor`, where :math:`\text{offset} + O_d \in [1, \text{n_block_wires} - 1]`
```

Comment:
We could also have the offset not be relative to `O_d`. I found it a bit confusing to parse what this line in the docstrings means.